markus_test_scripts/r_helpers.R

```diff
@@ -0,0 +1,316 @@
+#!/usr/bin/env Rscript
+
+# r_helpers.R - Core helper functions for RScript-based Markus testers
+# This file provides R equivalents of the Python llm_helpers.py functionality
+# NOTE: The ai-autograding-feedback-r package should be loaded before sourcing this file
+
+suppressWarnings(suppressMessages({
+  library(jsonlite)
+  library(testthat)
+}))
+
+# Global constants
+ANNOTATION_PROMPT <- "These are the student mistakes you previously identified in the
+last message. For each of the mistakes you identified, return a JSON object containing
+an array of annotations, referencing the student's submission file for line and column #s.
+Each annotation should include: filename: The name of the student's file. content:
+A short description of the mistake. line_start and line_end: The line number(s) where the
+mistake occurs. Ensure the JSON is valid and properly formatted. Here is a sample format
+of the json array to return: { \"annotations\": [{\"filename\": \"student_code.R\",
+\"content\": \"Variable 'x' is unused.\", \"line_start\": 5, \"line_end\": 5}]}.
+ONLY return the json object and nothing else. Make sure the line #s don't exceed
+the number of lines in the file. You can use markdown syntax in the annotation's content,
+especially when denoting code."
+
+MINIMUM_ANNOTATION_WIDTH <- 8
+
+#' Read prompt from file
+#'
+#' @param prompt_name Name of the prompt file (without extension)
+#' @param prompts_dir Directory containing prompt files
+#' @return String containing prompt content or NULL if file not found
+read_prompt_from_file <- function(prompt_name, prompts_dir = "examples/prompts") {
```

Comment:
This function really should not be necessary. Pass in the path to the ai feedback library, and let the library handle processing the prompt.